numpy/_core/numerictypes.py

```diff
@@ -223,12 +223,10 @@ def issctype(rep):
         return False
     try:
         res = obj2sctype(rep)
-        if res and res != object_:
-            return True
-        return False
     except Exception:
         return False
-
+    else:
+        return res and res != object_
```

Comment:
Oddly, this is one was not correct, it should `return bool()` if anything.

Reply:
Hmm, good point. I think there was another that replaced True/False, but by a comparison.

Reply:
Looks like `bool()` doesn't work, it converts the dtypes to `np.bool`, so the resulting return types are a mixture of 'bool' and 'np.bool', which aren't the same. Not sure if that is a bug or not :)

Reply:
I think this was a manual fix. Indeed, it should be:
```
        return bool(res) and res != object_
```

Reply:
The `SIM` rules are not enabled by default, by the way. These made sense to me, but they can be reverted.

Reply:
Fix in #26883. I just used the original and added a test.

Reply:
I guess there is an import of `np.bool` or `*` import somewhere...  I do think the builtin bool would work fine.  Could also have been `res is not None and res != np.object_` I think.
But doesn't matter, reverting is fine for now.